### PR TITLE
Lists UI Updates

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-820

## What changed (plus any additional context for devs)
1. Remove all instances of "add to list" button and replace with "add to favorites" button (except token expanded state, we are not showing favorites button there)
3. Remove AddTokenSheet

## Screen recordings / screenshots

https://github.com/rainbow-me/rainbow/assets/15272675/3dfe6f88-8856-43c7-9160-61ec91a60806

### CODE REVIEW

File: src/components/coin-row/ExchangeCoinRow.js
Comment (benisgold) [inline src/components/coin-row/ExchangeCoinRow.js:1]: This file component doesn't seem like it's being used at all so removing it

File: src/screens/CurrencySelectModal.tsx
Comment (benisgold) [inline src/screens/CurrencySelectModal.tsx:1]: none of the "add to favorites" logic in this file is being used any longer so i removed it. this logic solely exists here: https://github.com/rainbow-me/rainbow/blob/@benisgold/lists4/src/components/exchange/ExchangeAssetList.tsx#L296

**Review by skylarbarrera — APPROVED** [review]
🖖

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**